helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -1261,27 +1260,46 @@ public void addClusterToGrandCluster(String clusterName, String grandCluster) {
       throw new HelixException("Cluster " + clusterName + " is not setup yet");
     }
 
-    IdealState idealState = new IdealState(clusterName);
+    ZKHelixDataAccessor accessor =
+        new ZKHelixDataAccessor(grandCluster, new ZkBaseDataAccessor<>(_zkClient));
+    PropertyKey.Builder keyBuilder = accessor.keyBuilder();
+
+    ClusterConfig grandClusterConfig = accessor.getProperty(keyBuilder.clusterConfig());
+    RebalanceConfig superClusterRebalanceConfig =
+        grandClusterConfig.getSuperClusterRebalanceConfig();
+
+    // Note that the other rebalance configurations that are not verified below will be ignored.
```

Comment:
I'm confused about this comment here. Does the following check validate all possible scenarios or not? Do you mean "other configuration" as valid or invalid?

Reply:
It is not. Since we reused the existing structure RebalanceConfig, there are several fields that are not applicable.
That is also why I put the check in the API instead of the config class.